### PR TITLE
+gnu.org/emacs

### LINKED ISSUE #336: +emacs (296/548)
Formulae: https://formulae.brew.sh/formula/emacs

### PR DESCRIPTION
**Pull request by getneil**
resolves #336

### CODE REVIEW

**Comment by getneil** [comment]
need help not sure what the issue is. its building correctly in my local arm64, but the x86-64 is failing: https://github.com/teaxyz/pantry/actions/runs/4478716501/jobs/7871840789?pr=827

**Comment by getneil** [comment]
@Borlaag you might have some ideas?

File: projects/gnu.org/emacs/package.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  invisible-island.net/ncurses: ^6.3.0
+  libjpeg-turbo.org: ^2.1.5.1
+  gnu.org/sed: ^4.9.0
+  gnu.org/texinfo: ^7.0.1
+  gnu.org/libidn2: ^2.3
+
+build:
+  dependencies:
+    gnu.org/autoconf: ^2.17.0
+    gnutls.org: ^3.6
```
Comment (mxcl) [inline projects/gnu.org/emacs/package.yml]: this is a runtime dep per the error messages

**Comment by mxcl** [comment]
I made a number of minor changes, though some were important. Worth checking the diff: https://github.com/teaxyz/pantry/pull/827/commits/9d55738a1c2b939675c62de7ea104945e6765877

**Comment by mxcl** [comment]
The issues are now relocatability issues. This file is the build path:

https://github.com/teaxyz/pantry/actions/runs/4484799057/jobs/7885721582?pr=827#step:9:147

For tests we move the tea prefix to expose these issues. tea packages must be relocatable.

**Comment by mxcl** [comment]
Creating the folders is not going to fix it. The problem is the package has hard coded paths to where it is built on the build machine. You have to edit emacs itself to be relocatable, or potentially emacs offers a relocatability flag to its configure.

Step into the build directory on your local machine and do `./configure --help`

**Comment by getneil** [comment]
> Creating the folders is not going to fix it. The problem is the package has hard coded paths to where it is built on the build machine. You have to edit emacs itself to be relocatable, or potentially emacs offers a relocatability flag to its configure.
> 
> Step into the build directory on your local machine and do `./configure --help`



> Creating the folders is not going to fix it. The problem is the package has hard coded paths to where it is built on the build machine. You have to edit emacs itself to be relocatable, or potentially emacs offers a relocatability flag to its configure.
> 
> Step into the build directory on your local machine and do `./configure --help`

will try that. thanks

**Comment by mxcl** [comment]
image

So yeah relocatability. I was hoping the env vars will help.

I'm reading docs and the sources as well as googling and using chatGPT to try and fix this.

You can't fix it by passing paths to `./configure` that will then be hardcoded to the build systems build prefixes…

**Comment by mxcl** [comment]
managed to fix with env vars, as said my strategy was: read the docs

**Pull request merged**